Ginger/GingerCoreNETUnitTest/RunTestslib/VariableInputOutoutMappingTest.cs

```diff
@@ -93,10 +93,30 @@ public void OutputVariableOldMappingValueRunsetConfigLoadTest()
 
             //Assert
             Assert.AreEqual("2887671c-4f0c-4be8-9233-ca2f20ed1f04_b3ed569d-c99b-4a5e-8bb3-1462d83a837f", runSetConfig.GingerRunners[0].BusinessFlowsRunList[2].BusinessFlowCustomizedRunVariables[0].MappedOutputValue, "outputvariables with same name from different flow");
-            Assert.AreEqual("d1677d32-b215-4fd2-82aa-1d64c92317ab_b3ed569d-c99b-4a5e-8bb3-1462d83a837f", runSetConfig.GingerRunners[1].BusinessFlowsRunList[2].BusinessFlowCustomizedRunVariables[0].MappedOutputValue,  "outputvariables with same name from different instances of same flow");
-            Assert.AreEqual("d1677d32-b215-4fd2-82aa-1d64c92317ab_b3ed569d-c99b-4a5e-8bb3-1462d83a837f", runSetConfig.GingerRunners[2].BusinessFlowsRunList[0].BusinessFlowCustomizedRunVariables[0].MappedOutputValue,"outputvariables with same name from different instances of same flow from prev runner");
+            Assert.AreEqual("d1677d32-b215-4fd2-82aa-1d64c92317ab_b3ed569d-c99b-4a5e-8bb3-1462d83a837f", runSetConfig.GingerRunners[1].BusinessFlowsRunList[2].BusinessFlowCustomizedRunVariables[0].MappedOutputValue, "outputvariables with same name from different instances of same flow");
+            Assert.AreEqual("d1677d32-b215-4fd2-82aa-1d64c92317ab_b3ed569d-c99b-4a5e-8bb3-1462d83a837f", runSetConfig.GingerRunners[2].BusinessFlowsRunList[0].BusinessFlowCustomizedRunVariables[0].MappedOutputValue, "outputvariables with same name from different instances of same flow from prev runner");
 
         }
 
+        [TestMethod]
+        [Timeout(60000)]
+        public void InputVariableChangeFromSourceTest()
+        {
+            //Arrange         
+            ObservableList<BusinessFlow> businessFlows = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<BusinessFlow>();
+            BusinessFlow targetBusinessFlow = businessFlows.Where(b => b.Name.Contains("Check Variables")).FirstOrDefault();
+            ObservableList<GingerCore.Variables.VariableBase> lstVariables = targetBusinessFlow.Activities[0].Variables;
+            //Act
+            RunSetConfig runSetConfig = (from x in WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<RunSetConfig>() where x.Name == "Check Variables Run Set" select x).SingleOrDefault();
+            WorkSpace.Instance.RunsetExecutor.RunSetConfig = runSetConfig;
+            WorkSpace.Instance.RunsetExecutor.RunsetExecutionEnvironment = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>().FirstOrDefault();
+            WorkSpace.Instance.RunsetExecutor.InitRunners();
+            ObservableList<GingerCore.Variables.VariableBase> lstRunSetVariables = runSetConfig.GingerRunners[0].BusinessFlowsRunList[0].BusinessFlowCustomizedRunVariables;
+            
+            //Assert
+            Assert.AreEqual(null, lstRunSetVariables[0].Value, "Var A is supposed to be equal to null");
```

Comment:
we agreed to add 3 variables right ?
1 different from origin and null value
2. Different from origin and non null value
3. Same as origin

for the 2nd use case we are missing the assert

Reply:
For 2nd case I have added Assert.AreNotEqual on line 118, for 3rd case, I found that variables which are not changed from origin are not stored within RunSet Variables. Hence, thought to skip that.